modules/redis/scanner.go

```diff
@@ -152,6 +210,71 @@ func (scan *scan) Close() {
 	defer scan.close()
 }
 
+func getUnmarshaler(file string) (func([]byte, interface{}) error, error) {
+	var unmarshaler func([]byte, interface{}) error
+	switch ext := filepath.Ext(file); ext {
+	case ".json":
+		unmarshaler = json.Unmarshal
+	case ".yaml", ".yml":
+		unmarshaler = yaml.Unmarshal
+	default:
+		err := fmt.Errorf("File type %s not valid.", ext)
+		return nil, err
+	}
+	return unmarshaler, nil
+}
+
+func getFileContents(file string, output interface{}) error {
+	unmarshaler, err := getUnmarshaler(file)
+	if err != nil {
+		return err
+	}
+	fileContent, err := ioutil.ReadFile(file)
```

Comment:
since `ReadFile` reads the whole file into RAM, should we check file size first, and err out if it is too large (by some arbitrary number)?

Reply:
Since these are command line flags, I assumed when adding this that we wouldn't be malicious to ourselves, so we wouldn't add files that are too obnoxiously large

Reply:
Tend to agree, any user input is probably trusted, the exception being anything that could wreak havoc on the hosts we scan or on our scanning infrastructure

Reply:
Generally speaking, that goes against traditional defensive programming practices -- all user input is untrusted, including files.  The mistake we programmers often make is thinking "no one will use this program differently than it is used now", or to always assume the filesystem can be trusted (or that mistakes don't happen!).

Reply:
I can add the check -- what would be a good limit for the file size?

Reply:
I guess the "custom commands" file would be the larger of the two you are reading... how many custom commands are you traditionally expecting?  My guess is pretty low, so maybe an order or two of magnitude higher than that (100k?  Would you expect custom commands to ever get close to that?).

Reply:
I'd say 100KB is a pretty large upper limit -- definitely at least a magnitude or two above what I would expect.

Reply:
That being said, I made the upper limit 100KB and added a command-line flag to raise or lower that limit